staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -348,26 +384,26 @@ func (b *builder) buildRoute(root, path, httpMethod, actionVerb, operationVerb s
 
 // buildKubeNative builds input schema with Kubernetes' native object meta, type meta and
 // extensions
-func (b *builder) buildKubeNative(schema *structuralschema.Structural, v2 bool, crdPreserveUnknownFields bool) (ret *spec.Schema) {
+func (b *builder) buildKubeNative(schema *structuralschema.Structural, opts Options, crdPreserveUnknownFields bool) (ret *spec.Schema) {
 	// only add properties if we have a schema. Otherwise, kubectl would (wrongly) assume additionalProperties=false
 	// and forbid anything outside of apiVersion, kind and metadata. We have to fix kubectl to stop doing this, e.g. by
 	// adding additionalProperties=true support to explicitly allow additional fields.
 	// TODO: fix kubectl to understand additionalProperties=true
-	if schema == nil || (v2 && (schema.XPreserveUnknownFields || crdPreserveUnknownFields)) {
+	if schema == nil || ((opts.V2 && !opts.OverrideToStructuralOpenAPIV2) && (schema.XPreserveUnknownFields || crdPreserveUnknownFields)) {
 		ret = &spec.Schema{
 			SchemaProps: spec.SchemaProps{Type: []string{"object"}},
 		}
 		// no, we cannot add more properties here, not even TypeMeta/ObjectMeta because kubectl will complain about
 		// unknown fields for anything else.
 	} else {
-		if v2 {
+		if opts.V2 && !opts.OverrideToStructuralOpenAPIV2 {
```

Comment:
OverrideToStructuralOpenAPIV2 is really `filterSchemaForKubectlOpenapiV2Validation`. Can we rename that?

Reply:
Renamed to `SkipFilterSchemaForKubectlOpenAPIV2Validation` because we want the inverse boolean value. Without inversing the boolean, the code would need to change in a lot more places (Changing SSA usage vs change all usage other than SSA)